File: Python/004.py
from Utils import isPalindrom

res = 0
for i in reversed(range(100,1000)):
    if i * 999 <= res:
        break
    for j in reversed(range(i,1000)):
        product = i * j
        if product <= res:
            break
        if isPalindrom(str(product)):
            res = product
print(res)

File: C++/004.cpp
#include <iostream>
#include "Utils.h"

int main(){
  int res = 0;
  int product;
  for (int i=999; i>99; i--){
    if (i*999 <= res){
      break;
    }
    for (int j=999; j>=i; j--){
      product = i*j;
      if (product<=res){
        break;
      }
      if (isPalindrom(product)){
        res = product;
      }
    }
  }
  cout << res;
}
